modules/ztp-configuring-ptp-fast-events.adoc

```diff
@@ -0,0 +1,92 @@
+
+You can configure PTP fast events for vRAN clusters that are deployed using the GitOps Zero Touch Provisioning (ZTP) pipeline. Use `PolicyGenTemplate` custom resources (CRs) as the basis to create a hierarchy of configuration files tailored to your specific site requirements.
+
+The `PolicyGenTemplate` CRs that are relevant to PTP events can be found in the `/home/ztp/Argo cd/example` folder in the `quay.io/redhat_emp1/ztp-site-generator:latest` reference architecture container image. The reference architecture has a `/policygentemplates` and `/siteconfig` folder. The `/policygentemplates` folder has common, group, and site-specific configuration CRs. Each `PolicyGenTemplate` CR refers to other CRs that are in the `/source-crs` folder of the reference architecture.
+
+The `PolicyGenTemplate` CRs required to deploy PTP fast events are described below.
+
+.PolicyGenTemplate CRs for vRAN deployments
+[cols=2*, options="header"]
+|====
+|PolicyGenTemplate CR
+|Description
+
+|`common-ranGen.yaml`
+|Contains the common RAN policies that get applied to all clusters. To deploy the PTP Operator to your clusters, you configure `Namespace`, `Subscription`, and `OperatorGroup` CRs.
+
+|`group-du-3node-ranGen.yaml`
+|Contains the RAN policies for three-node clusters only, including PTP fast events configuration.
+
+|`group-du-sno-ranGen.yaml`
+|Contains the RAN policies for single-node clusters only, including PTP fast events configuration.
+
+|`group-du-standard-ranGen.yaml`
+|Contains the RAN policies for standard three control-plane clusters, including PTP fast events configuration.
+|====
+
+.Prerequisites
+
+* Create a git repository where you manage your custom site configuration data.
+* Extract the contents of the `/home/ztp` folder from the `quay.io/redhat_emp1/ztp-site-generator:latest` reference architecture container image, and review the changes.
+
+.Procedure
+
+. Add the following YAML into `.spec.sourceFiles` in the `common-ranGen.yaml` file to configure the AMQP Operator:
++
+[source,yaml]
+----
+#AMQ interconnect operator for fast events
+- fileName: AmqSubscriptionNS.yaml
+  policyName: "amq-sub-policy"
+- fileName: AmqSubscriptionOperGroup.yaml
+  policyName: "amq-sub-policy"
+- fileName: AmqSubscription.yaml
+  policyName: "amq-sub-policy"
+----
+
+. Apply the following `PolicyGenTemplate` changes to `group-du-3node-ranGen.yaml`, `group-du-sno-ranGen.yaml`, or `group-du-standard-ranGen.yaml` files according to your requirements:
+
+.. In `.sourceFiles`, change the `config-policy` to reference the `PtpOperatorConfig` CR that configures the AMQ transport host:
++
+[source,yaml]
+----
+- fileName: PtpOperatorConfigForEvent.yaml
+  policyName: "config-policy"
+----
+
+.. Configure the `linuxptp` and `phc2sys` for the PTP clock type and interface. For example, add the following stanza into `.sourceFiles`:
++
+[source,yaml]
+----
+- fileName: PtpConfigSlave.yaml                   <1>
+  policyName: "ptp-config-policy"
```

Comment:
We should use "config-policy" instead of "ptp-config-policy" to minimize ztp time. Note that a policy can include multiple CRs as shown in ZTP examples.